mscore/resourceManager.cpp

```diff
@@ -264,6 +414,343 @@ bool ResourceManager::verifyLanguageFile(QString filename, QString hash)
       return verifyFile(local, hash);
       }
 
+//---------------------------------------------------------
+//   GitHub utility functions
+//---------------------------------------------------------
+static inline QString githubLatestReleaseAPI(const QString &user, const QString &repo) { return "https://api.github.com/repos/" + user + "/" + repo + "/releases"; }
+static inline QString githubCommitAPI(const QString &user, const QString &repo, const QString &ref = "")
+      {
+      return "https://api.github.com/repos/" + user + "/" + repo + "/commits" + (ref.isEmpty() ? "" : ('/' + ref));
+      }
+static inline QString githubLatestArchiveURL(const QString &user, const QString &repo, const QString &branch = "master")
+      {
+      //return "https://api.github.com/repos/" + user + "/" + repo + "/zip/" + branch;
+      return "https://github.com/" + user + '/' + repo + "/archive/" + branch + ".zip";
+      }
+static QString getLatestCommitSha(const QString& user, const QString &repo, const QString &branch = "master")
+      {
+      QString api_url = githubCommitAPI(user, repo, branch);
+      DownloadUtils json;
+      json.setTarget(api_url);
+      json.download();
+      QJsonDocument json_doc = QJsonDocument::fromJson(json.returnData());
+      QJsonObject latest_commit;
+      if (json_doc.isArray())
+            // there are  multiple commits, fetch the latest
+            latest_commit = json_doc.array().first().toObject();
+      else
+            latest_commit = json_doc.object();
+      if(latest_commit.contains("sha"))
```

Comment:
space bwteeen if and (

Reply:
Resolved in 77701599820befd5e5381af2075d63e6b5b3dd85.